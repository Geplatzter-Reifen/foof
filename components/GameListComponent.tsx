import { BoardGame } from "@/model/model";
import { View, Text } from "react-native";

export default function GameListComponent({ games }: { games: BoardGame[] }) {
  return (
    <View>
      {games.map((game) => (
        <View key={game.title}>
          <Text>{game.title} - </Text>
          <Text>Min Players: {game.minPlayers}</Text>
        </View>
      ))}
    </View>
  );
}
