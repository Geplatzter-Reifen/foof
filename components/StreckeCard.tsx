import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import {DATE_TIME, dateFormat} from "@/utils/dateFormatter";

export type StreckeData = {
  startLoc: string,
  endLoc: string,
  startTime: Date,
  endTime: Date,
}

export default function StreckeCard(props: StreckeData) {
  return (
    <View style={styles.card}>
      <Text style={styles.title}>Strecke</Text>
      <Text style={styles.label}>Start:</Text>
      <Text style={styles.info}>{props.startLoc}</Text>
      <Text style={styles.label}>Ziel:</Text>
      <Text style={styles.info}>{props.endLoc}</Text>
      <Text style={styles.label}>Startzeitpunkt:</Text>
      <Text style={styles.info}>{dateFormat(props.startTime, DATE_TIME)}</Text>
      <Text style={styles.label}>Endzeitpunkt:</Text>
      <Text style={styles.info}>{dateFormat(props.endTime, DATE_TIME)}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  card: {
    backgroundColor: '#f5f5f5',
    width: '90%',
    padding: 20,
    marginVertical: 10,
    borderRadius: 10,
    shadowColor: '#000',
    shadowOffset: {width: 0, height: 2},
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 2,
  },
  title: {
    fontSize: 18,
    fontWeight: 'bold',
    color: 'green',
    marginBottom: 10,
  },
  label: {
    fontSize: 14,
    fontWeight: 'bold',
    color: '#333',
    marginTop: 5,
  },
  info: {
    fontSize: 16,
    color: '#666',
  },
});