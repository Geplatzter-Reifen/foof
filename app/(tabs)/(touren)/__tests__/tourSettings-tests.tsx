import React from "react";
import { render, cleanup } from "@testing-library/react-native";
import { ApplicationProvider } from "@ui-kitten/components";
import * as eva from "@eva-design/eva";
import TourSettings from "./../tourSettings"; // Adjust the path as needed

// Mock dependencies
jest.mock("react-native-safe-area-context", () => ({
  useSafeAreaInsets: jest.fn(() => ({
    top: 10,
    bottom: 0,
    left: 0,
    right: 0,
  })),
}));

jest.mock("@ui-kitten/components", () => {
  const originalModule = jest.requireActual("@ui-kitten/components");
  return {
    ...originalModule,
    Icon: jest.fn().mockImplementation(({ name }) => `MockedIcon(${name})`),
  };
});

jest.mock("@/components/TopNavigation/renderBackAction", () => jest.fn());
jest.mock(
  "@/components/Settings/TourNameSettingsSection",
  () => "TourNameSettingsSection",
);
jest.mock(
  "@/components/Settings/RouteSettingsSection",
  () => "RouteSettingsSection",
);
jest.mock("@/constants/styles", () => ({
  basicCard: { borderRadius: 8 },
  basicShadow: { shadowColor: "#000" },
}));

afterEach(() => {
  jest.clearAllMocks();
  cleanup();
});

describe("TourSettings", () => {
  it("renders correctly", () => {
    const { toJSON } = render(
      <>
        {/* Provide MaterialIconsPack */}
        <ApplicationProvider {...eva} theme={eva.light}>
          <TourSettings />
        </ApplicationProvider>
      </>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it("renders the first Layout at depth 1", () => {
    const { getAllByTestId } = render(
      <ApplicationProvider {...eva} theme={eva.light}>
        <TourSettings />
      </ApplicationProvider>,
    );

    // Query all Layout components by testID
    const layouts = getAllByTestId("layout"); //testID="layout"
    expect(layouts).toHaveLength(3); // Ensure there are three Layouts
  });

  it("renders a Card with rounded corners", () => {
    const { getByTestId } = render(
      <ApplicationProvider {...eva} theme={eva.light}>
        <TourSettings />
      </ApplicationProvider>,
    );

    const card = getByTestId("card"); // Add testID="card"
    expect(card.props.style).toEqual(
      expect.objectContaining({
        borderRadius: 8, // Ensure the card has rounded corners
        marginTop: 10,
        alignSelf: "flex-start",
        paddingHorizontal: 15,
        marginHorizontal: 10,
      }),
    );
  });
});
