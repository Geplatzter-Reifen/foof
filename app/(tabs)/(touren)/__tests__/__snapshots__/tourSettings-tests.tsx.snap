// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TourSettings renders correctly 1`] = `
<View
  appearance="default"
  level="1"
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "flex": 1,
      },
    ]
  }
  testID="layout"
>
  <View
    appearance="default"
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "minHeight": 56,
          "paddingHorizontal": 8,
          "paddingVertical": 8,
        },
        {
          "alignItems": "center",
          "flexDirection": "row",
        },
        {
          "justifyContent": "space-between",
        },
        {
          "marginTop": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "zIndex": 1,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Text
        appearance="default"
        category="h4"
        status="basic"
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 26,
              "fontWeight": "800",
            },
            undefined,
          ]
        }
      >
        Toureinstellungen
      </Text>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "row",
            "zIndex": 1,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    appearance="default"
    style={
      [
        {
          "backgroundColor": "#b8b8b8",
          "height": 1,
        },
        undefined,
      ]
    }
  />
  <View
    appearance="default"
    level="2"
    style={
      [
        {
          "backgroundColor": "#eaeaea",
        },
        {
          "flex": 1,
        },
      ]
    }
    testID="layout"
  >
    <View
      appearance="default"
      level="2"
      style={
        [
          {
            "backgroundColor": "#eaeaea",
          },
          {
            "flex": 1,
          },
        ]
      }
      testID="layout"
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "flex-start",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#7e7e7e",
            "borderRadius": 8,
            "borderWidth": 1,
            "marginHorizontal": 10,
            "marginTop": 10,
            "opacity": 1,
            "overflow": "hidden",
            "paddingHorizontal": 15,
            "shadowColor": "#000",
          }
        }
        testID="card"
      >
        <View
          style={
            {
              "backgroundColor": undefined,
              "height": 0,
            }
          }
        />
        <View
          style={
            [
              {
                "flexShrink": 1,
              },
              {
                "paddingHorizontal": 24,
                "paddingVertical": 16,
              },
            ]
          }
        >
          <TourNameSettingsSection />
          <RouteSettingsSection />
        </View>
      </View>
    </View>
  </View>
</View>
`;
